composer/callbacks/checkpoint_saver.py

```diff
@@ -0,0 +1,439 @@
+            If ``False`` (the default), then the ``checkpoint_folder`` must not exist or be empty.
+            (default: ``False``)
+
+        should_save (Time | str | int | (State, Event) -> bool): A :class:`Time`, time-string, integer (in epochs),
+            or a function that takes (state, event) and returns a boolean whether a checkpoint should be saved.
+
+            If an integer, checkpoints will be saved every n epochs.
+            If :class:`Time` or a time-string, checkpoints will be saved according to this interval.
+
+            .. seealso:: :func:`.checkpoint_periodically`
+
+            If a function, then this function should take two arguments (:class:`State`, :class:`Event`).
+            The first argument will be the current state of the trainer, and the second argument will be
+            be :attr:`.Event.BATCH_CHECKPOINT` or :attr:`.EPOCH_CHECKPOINT` (depending on the current training
+            progress). It should return ``True`` if a checkpoint should be saved given the current state and
+            event.
+
+        weights_only (bool): If ``True``, save only the model weights instead of the entire training state.
+            This parmeter must be ``False`` when using DeepSpeed. (default: ``False``)
+
+    Attributes:
+        checkpoint_folder (str): The folder in which checkpoints are stored. If an absolute path was specified for
+            ``save_folder`` upon instantiation, then that path will be used. Otherwise, this folder is relative to
+            the run directory of the training run (e.g. ``{run_directory}/{save_folder}``).
+            If no run directory is provided, then by default, it is of the form
+            ``runs/<timestamp>/rank_<GLOBAL_RANK>/<save_folder>`` where ``timestamp``
+            is the start time of the run in iso-format, ``GLOBAL_RANK`` is the global rank of the process,
+            and ``save_folder`` is the save_folder argument provided upon construction.
+
+            .. seealso:: :mod:`~composer.utils.run_directory` for details on the format of the run directory
+                and how to customize it.
+        saved_checkpoints (Dict[Timestamp, List[str]]): A dictionary mapping a save timestamp
+            to a list of filepaths corresponding to the checkpoints saved at that time.
+
+            .. note:: When using DeepSpeed, the index of a filepath in each list corresponds to the
+                global rank of the process that wrote that file. These filepaths are valid only on
+                the global rank's node. Otherwise, when not using DeepSpeed, this list will contain
+                only one filepath since only rank zero saves checkpoints.
+    """
+
+    def __init__(
+        self,
+        save_folder: str = "checkpoints",
+        name_format_string: str = "ep{epoch}-ba{batch}/rank_{rank}",
+        latest_symlink_format_string: Optional[str] = "latest/rank_{rank}",
```

Comment:
{timestamp}/latest.pt

Reply:
Renamed to `latest-rank{rank}` per offline discussion